src/services/catalog.service.ts

```diff
@@ -21,26 +21,42 @@ export class CatalogService {
   }
 
   public getCatalogItems(includeTemplates: boolean) {
-    let promises: any = {
-      serviceClasses: this.dataService.list({
-        group: 'servicecatalog.k8s.io',
-        resource: 'serviceclasses'
-      }, {}),
-      imageStreams: this.dataService.list("imagestreams", { namespace: "openshift" })
-    };
+    let deferred = this.$q.defer();
+    let catalogItems: any = {};
+    let totalNumPromises: number = includeTemplates ? 3 : 2;
+    let numPromisesExecuted: number = 0;
+    let errorMsg: string = '';
+
+    this.dataService.list({
+      group: 'servicecatalog.k8s.io',
+      resource: 'serviceclasses'
+    }, {}).then( (resources: any) => {
+      catalogItems.serviceClasses = resources.by("metadata.name");
+    }, () => {
+      errorMsg += 'Error loading service classes.';
+    }).finally(() => {
+      this.returnCatalogItems(deferred, catalogItems, ++numPromisesExecuted, totalNumPromises, errorMsg);
+    });
+
+    this.dataService.list("imagestreams", {namespace: "openshift"}).then( (resources: any) => {
+      catalogItems.imageStreams = resources.by("metadata.name");
+    }, () => {
+      errorMsg += 'Error loading OpenShift imagestreams. ';
+    }).finally(() => {
+      this.returnCatalogItems(deferred, catalogItems, ++numPromisesExecuted, totalNumPromises, errorMsg);
+    });
 
     if (includeTemplates) {
-      promises.templates = this.dataService.list("templates", { namespace: "openshift" });
+      this.dataService.list("templates", {namespace: "openshift"}).then((resources: any) => {
+        catalogItems.templates = resources.by("metadata.name");
+      }, () => {
+        errorMsg += 'Error loading Templates.';
```

Comment:
also if you have multiple failures the messages are going to have no spaces between them, I recommend keeping an errorMsgs array, and then doing an array join(' ') at the end